### PR TITLE
Ensure that labels for different k8s targets don't interfere with each other

### LINKED ISSUE #26516: Kubernetes extension(s) have stopped generating the app.openshift.io/runtime label
### Describe the bug

I'm not sure which extension it is, but since Quarkus 2.10.1.Final the `app.openshift.io/runtime: quarkus` label is now missing from resources generated when using `quarkus.kubernetes.deployment-target=openshift`. For some reason it is still working when setting `quarkus.kubernetes.deployment-target=knative`.

You can see this in this diff: https://github.com/quarkusio/quarkus-super-heroes/commit/fe02dae8e887df95e903bda3b5a36baf6f075336?diff=unified#diff-dc6194623579af548ca3da8872f2c72c20d41fb010026243a203b2d52da3defaL48

This commit is the diff between `2.10.0.Final` & `2.10.1.Final`. There were no other commits in between the upgrade to `2.10.0.Final` (where it was behaving correctly) and `2.10.1.Final` (where it is missing).

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #26516

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java [hunk 1]
```diff
@@ -192,6 +193,7 @@ public List<DecoratorBuildItem> createDecorators(ApplicationInfoBuildItem applic
         config.maxScale.map(String::valueOf).ifPresent(max -> result.add(new DecoratorBuildItem(KNATIVE,
                 new ApplyAnnotationsToServiceTemplate(name, KNATIVE_MAX_SCALE, max))));
         /**
+         * Βσ
```
Comment (gsmet) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java:196]: ?

Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java:196]: LOL. Who doesn't want random greek characters?

Comment (edeandrea) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeProcessor.java:196]: Its all greek to me :D

**Comment by geoand** [comment]
@iocanel can you take a look at this one please?

Thanks

**Review by iocanel — APPROVED** [review]
LGTM

**Pull request merged**